Mathlib/GroupTheory/GroupAction/Amenable.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2023 Matthias Uschold. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matthias Uschold
+-/
+import Mathlib.GroupTheory.GroupAction.Defs
+import Mathlib.Data.Real.ENNReal
+import Mathlib.MeasureTheory.MeasurableSpace.Defs
+
+
+/-!
+# Amenable Monoid Actions
+
+In this file, we define amenability of a monoid action.
+A monoid action is amenable if it admits an invariant mean.
+A mean is like a probability measure, demanding
+only finite additivity instead of σ-additivity.
+
+## Main Definitions
+
+- `Mean`: defines means as finitely additive probability measures
+- `InvariantMean`: defines when a mean is invariant under a monoid action
+- `amenable`: A monoid action is amenable if there exists an invariant mean
+
+## Implementation Notes
+
+`Mean` was not implemented using MeasureTheory.ProbabilityMeasure
+because meausures are σ-additive (i.e. countably additive). In this
+setting, this would be a too strong assumption, we only want to demand
+finite additivity. Typically, the resulting measures will not be
+σ-additive.
+
+## References
+
+See [bartholdi2017amenability] for definitions
+and more information on amenable actions.
+
+-/
+
+universe u v
+variable (α : Type v) [MeasurableSpace α]
+
+/--A mean is a function from the power set of α to ENNReal that
+- assigns the value 1 to the full set α, and
+- is finitely additive under disjoint unions -/
+structure Mean where
+  /-- function giving the measure of a measurable subset-/
+  measureOf : {S // MeasurableSet (α := α) S} → NNReal
+  /-- Set.univ has measure 1  -/
+  measureOf_univ : measureOf ⟨Set.univ, MeasurableSet.univ⟩ = 1
+  /-- measureOf is finitely additive -/
+  fin_add : ∀ (X Y : Set α),
+      (hX : MeasurableSet X) → (hY: MeasurableSet Y) → Disjoint X Y
+      → measureOf (⟨X ∪ Y, MeasurableSet.union hX hY⟩) = measureOf ⟨X, hX⟩ + measureOf ⟨Y, hY⟩
+
+@[coe]
+instance : CoeFun (Mean α) (λ _ => {S // MeasurableSet (α := α) S} → NNReal) where
+  coe := Mean.measureOf
+
+
+variable (G : Type u) [Monoid G] [MulAction G α]
+  [MulActionMeasurable : Fact (∀ (g : G), Measurable (λ (x : α) => g • x))]
```

Comment:
I assume you're aware of [`MeasurableSMul`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Group/Arithmetic.html#MeasurableSMul)? This makes it look like you maybe want a weaker `MeasurableConstSMul`, to match all the other `*ConstSMul` typeclasses.

Reply:
For now, I think it makes sense to assume `MeasurableSMul`, and leave a todo note that it could be extended to `MeasurableConstSMul` once this is defined.